File: PersonalAssistant.DataAccess/Entities/Assignment.cs
using System;
using System.Collections.Generic;
using PersonalAssistant.DataAccess.Identity;

namespace PersonalAssistant.DataAccess.Entities
{
    public class Assignment
    {
        public int Id { get; set; }

        public int? DisciplineId { get; set; }

        public Discipline Discipline { get; set; }

        public int? AssignmentTypeId { get; set; }

        public AssignmentType AssignmentType { get; set; }

        public int? CreatorId { get; set; }

        public User Creator { get; set; }

        public int? ExecutorId { get; set; }

        public User Executor { get; set; }

        public string TopicName { get; set; }

        public DateTime PreferredDeadline { get; set; }

        public DateTime CreationDate { get; set; }

        public string Details { get; set; }

        public IEnumerable<Comment> Comments { get; set; }

        public IEnumerable<File> Files { get; set; }
    }
}

File: PersonalAssistantUI/src/app/models/assignment.ts
import { AssignmentType } from "./assignment-type";
import { Discipline } from "./discipline";
import { User } from "./user/user";

export class Assignment {
    public id: number;
    public creatorId: number;
    public creator: User;
    public executorId: number;
    public executor: User;
    public topicName: string;
    public preferredDeadline: Date;
    public disciplineId: number;
    public discipline: Discipline;
    public assignmentTypeId: number;
    public assignmentType: AssignmentType;
    public details: string;

    constructor() {
        this.id = 0;
        this.creatorId = null;
        this.executorId = null;
        this.topicName = '';
        this.preferredDeadline = new Date();
        this.disciplineId = null;
        this.discipline = null;
        this.assignmentTypeId = null;
        this.assignmentType = null;
        this.details = '';
    }
}